services/api-server/tests/unit/api_solvers/test_api_routers_solvers_jobs_logs.py

```diff
@@ -123,3 +123,49 @@ async def test_log_streaming(
         collected_messages
         == fake_log_distributor._produced_logs[: len(collected_messages)]
     )
+
+
+@pytest.fixture
+async def mock_job_not_found(
+    mocked_directorv2_service_api_base,
+) -> AsyncIterable[MockRouter]:
+    def _get_computation(request: httpx.Request, **kwargs) -> httpx.Response:
+        return httpx.Response(status_code=status.HTTP_404_NOT_FOUND)
+
+    mocked_directorv2_service_api_base.get(
+        path__regex=r"/v2/computations/(?P<project_id>[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})"
+    ).mock(side_effect=_get_computation)
+    return mocked_directorv2_service_api_base
+
+
+async def test_logstreaming_job_not_found_exception(
+    app: FastAPI,
+    auth: httpx.BasicAuth,
+    client: httpx.AsyncClient,
+    solver_key: str,
+    solver_version: str,
+    fake_log_distributor,
+    fake_project_for_streaming: ProjectGet,
+    mock_job_not_found: MockRouter,
+):
+
+    job_id: JobID = fake_project_for_streaming.uuid
+    _received_error = False
+
+    async with client.stream(
+        "GET",
+        f"/{API_VTAG}/solvers/{solver_key}/releases/{solver_version}/jobs/{job_id}/logstream",
+        auth=auth,
+    ) as response:
+        response.raise_for_status()
+        async for line in response.aiter_lines():
+            try:
+                job_log = JobLog.parse_raw(line)
+                pprint(job_log.json())
+            except ValidationError:
+                error = ErrorGet.parse_raw(line)
```

Comment:
so you rely on ValidationError to "switch" between OK or KO... 
TIP: alternatively
```python
result = parse_raw_as( JobLog | ErrorGet, line)
if isinstance(result, ErrorGet):
    raise RuntimeError("something went wrong)
```
this way you can also use ValidationError to check the structure of the transmitted data